### DISCUSSION TITLE
findMany query returns null for relational tables ( many to many relation)

### DISCUSSION THREAD

**Discussion by rosnk** [prisma/prisma#4277] [Q&A] [answered]
i have created joining model for two many to many models in schema.prisma file. Also Have successfully migrated data in postgresql. 

Now i am trying to use following code to get data from model and also data from its relation table. I do get result but result contains joining table rather then relation table. 

Output from include field has null, i guess its because i am doing somethingwrong to say to include many to many relation model rather then including joining table. 
```

prisma.{model_name}.findMany({
include: {{relation_table_name}: true}
})
```

*{relation_table_name} -> placeholder for table defined in schema.prisma
{model_name}-> placehoder for modelname

Any help will be appreciated.

**Comment by ryands17** [comment, marked as answer]
@rosnk 👋 
Could you share your schema if possible?

**Reply by rosnk** [reply]
@ryands17 here are my code

schema.prisma


```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}



model Listing{
  id                    Int                   @id @default(autoincrement())
  title                 String
  description           String
  mobileHomeCategories  MobileHomeCategoryOnListings[]
}


model MobileHomeCategory{
  id                    Int                   @id @default(autoincrement())
  name                  String
  listings              MobileHomeCategoryOnListings[]
}

model MobileHomeCategoryOnListings{
  listing     Listing     @relation(fields: [listingId], references: [id])
  listingId   Int       
  mobileHomeCategory    MobileHomeCategory @relation(fields: [categoryId], references: [id])
  categoryId  Int      
  createdAt   DateTime @default(now())
  updatedAt   DateTime @default(now())
  @@id([listingId, categoryId])
}

```



app.js : apollo server configuration with express
```js
const express = require("express");
const { ApolloServer, gql } = require("apollo-server-express");

const { loadFile } = require("graphql-import-files");
const Mutation = require("./src/resolvers/Mutation");
const Query = require("./src/resolvers/Query");
const { PrismaClient } = require("@prisma/client");
const prisma = new PrismaClient();

const server = new ApolloServer({
  typeDefs: loadFile("./src/schema/schema.graphql"),
  resolvers: { Mutation, Query },
  context: { prisma },
});

const app = express();
server.applyMiddleware({ app });

app.listen({ port: 5000 }, () =>
  console.log(`🚀 Server ready at http://localhost:5000${server.graphqlPath}`)
);
```

typeDefs in apollo server is loaded using external file (line ->typeDefs: loadFile("./src/schema/schema.graphql"),): 


```graphql
type Query {
  mobileHomeCategories: [MobileHomeCategory]
}

type MobileHomeCategory {
  id: ID!
  name: String!
  listings: [Listing]
}

type Listing {
  id: ID
  title: String
  description: String
}
```


query.js loaded using external file in app.js: (line-> resolvers: { Mutation, Query }, )


```js
const Query = {
  mobileHomeCategories: async (parent, args, context, info) => {
    const mobileHomeCategories = await context.prisma.mobileHomeCategory.findMany(
      {
        include: { listings: true },
      }
    );
    console.log(mobileHomeCategories);
    return mobileHomeCategories;
  },
};

module.exports = Query;
```


Query: 


```graphql
query{
  mobileHomeCategories{
    id
    name
    listings{
      title
      description
    }
  }
}
```



Query Result: 

```json
{
  "data": {
    "mobileHomeCategories": [
      {
        "id": "1",
        "name": "Recent Listings",
        "listings": [
          {
            "title": null,
            "description": null
          },
          {
            "title": null,
            "description": null
          },
        ]
      }
}

```

**Reply by ryands17** [reply]
@rosnk As you have an explicit join table, you will need to query it in this way as per your schema:

```ts
  const data = await prisma.mobileHomeCategory.findMany({
    include: { listings: { include: { listing: true } } },
  })
```

But this data will include the listings in. a nested fashion due to the join table:
![image](https://user-images.githubusercontent.com/19697099/99360322-6bb3f400-28d6-11eb-98e1-77f01ef6c390.png)

To get the listings exactly as required, you would need to map the data and then send it:

```ts
const listings = data.map(mhc => {
    return {
      ...mhc,
      listings: mhc.listings.map(listing => listing.listing),
    }
})
```

This way, you will directly get the listings under `mobileHomeCategory`.

**Reply by rosnk** [reply]
@ryands17 thanks works perfectly. 

I wish prisma will incorporate this in their documentation, or may be they have, its just i did not come across it.

**Reply by ryands17** [reply]
@rosnk we do have a section [here](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#nested-reads) for nested includes for relation reads :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
